### PR TITLE
05923 Fixed persistence issue for a restored `LongListOffHeap`

### LINKED ISSUE #5923: `LongListOffHeap` restored from a snapshot persists data incorrectly
### Description

Due to a bug in the persist logic, `LongListOffHeap` incorrectly persists the data if it was restored from a snapshot.

### Steps to reproduce

1. Create an instance of `LongListOffHeap`. 
2. Fill it with values
3. Create a snapshot of it (call `AbstractLongList#writeToFile`)
4. Create another instance of the list using the snapshot
5. Create a snapshot of the restored list
6. Create one more instance using a list created on the previous step
7. Verify the list contents. 

### Additional context

_No response_

### Hedera network

other

### Version

v0.37-SNAPSHOT

### Operating system

None

### PR DESCRIPTION
**Pull request by imalygin**
**Description**:

Due to a bug in the persist logic, `LongListOffHeap` incorrectly persists the data if it was restored from a snapshot.

Steps to reproduce
- Create an instance of LongListOffHeap.
- Fill it with values
- Create a snapshot of it (call AbstractLongList#writeToFile)
- Create another instance of the list using the snapshot
- Create a snapshot of the restored list
- Create one more instance using a list created on the previous step
 - Verify the list contents.

**Related issue(s)**:

Fixes #5923 

**Notes for reviewer**:

**Checklist**

- Doesn't need additional domentation
- [x] The bug covered with `LongListOffHeapTest#updateListCreatedFromSnapshotPersistAndVerify` test

### CODE REVIEW

**Review by artemananiev — APPROVED** [review]

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListOffHeap.java [hunk 1]
```diff
@@ -169,7 +169,8 @@ protected void writeLongsData(final FileChannel fc) throws IOException {
             for (int i = firstChunkWithDataIndex; i < totalNumOfChunks; i++) {
                 final ByteBuffer byteBuffer = chunkList.get(i);
                 final ByteBuffer nonNullBuffer = requireNonNullElse(byteBuffer, emptyBuffer);
-                final ByteBuffer buf = nonNullBuffer.slice(); // slice so we don't mess with state
+                final ByteBuffer buf =
+                        nonNullBuffer.slice(0, nonNullBuffer.capacity()); // slice so we don't mess with state
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListOffHeap.java:173]: Not sure the comment is accurate.  The buffer returned from slice is still backed by the same internal data as the original (so writes to one are reflected in the other).

> Changes to this buffer's content will be visible in the new buffer, and vice versa

Comment (imalygin) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListOffHeap.java]: Fair enough. I updated the comment.

File: platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/collections/LongListOffHeapTest.java [hunk 2]
```diff
@@ -153,6 +153,35 @@ void testPersistShrunkList(final int chunkOffset) throws IOException {
         }
     }
 
+    @Test
+    void updateListCreatedFromSnapshotPersistAndVerify() throws IOException {
+        final int sampleSize = getSampleSize();
+        try (final LongListOffHeap list = createFullyParameterizedLongListWith(
+                sampleSize / 100, // 100 chunks, 100 longs each
+                sampleSize + DEFAULT_NUM_LONGS_PER_CHUNK)) {
+            for (int i = 1; i < getSampleSize(); i++) {
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/collections/LongListOffHeapTest.java:162]: Should this start at 0?  We're putting from 1-99, but below we're getting from 0 to 99?

**Review by jsync-swirlds — APPROVED** [review]

File: platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/collections/LongListOffHeapTest.java (hunk 2 above)
Comment (imalygin) [reply platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/collections/LongListOffHeapTest.java]: In this particular case it doesn't make any difference but I changed it for the sake of consistency.

**Review by artemananiev — APPROVED** [review]

**Review by OlegMazurov — APPROVED** [review]

**Pull request merged**